Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -165,17 +165,45 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
         {
-            if (boundaryCurves == null || boundaryCurves.Where(x => x != null).Count() == 0)
+            if (boundaryCurves == null || boundaryCurves.Count == 0 || boundaryCurves.All(x => x == null))
             {
-                BH.Engine.Reflection.Compute.RecordError("Cannot create planar surface from null curves.");
+                BH.Engine.Reflection.Compute.RecordError("Cannot create planar surface from a null or empty collection of boundary curves.");
                 return null;
             }
 
-            List<ICurve> checkedCurves = boundaryCurves.Where(x => x.IIsClosed(tolerance) && x.IIsPlanar(tolerance)).ToList();
+            //--------------Null-Boundary-Curves-----------------------//
+            int count = boundaryCurves.Count;
+            List<ICurve> checkedCurves = boundaryCurves.Where(x => x != null).ToList();
+            if (checkedCurves.Count != count)
+                Reflection.Compute.RecordWarning("Some of the input boundary curves were null and have been ignored on planar surface creation. Please make sure if the output is correct.");
+
+            //--------------Unsupported-Boundary-Curves-----------------------//
+            count = checkedCurves.Count;
+            checkedCurves = checkedCurves.Where(x => !(x is NurbsCurve) && !(x is Ellipse)).ToList();
+            if (checkedCurves.Count != count)
+                Reflection.Compute.RecordWarning("Some of the input boundary curves were nurbs curves or ellipses, which are unsupported, and have been ignored on planar surface creation. Please make sure if the output is correct.");
+
+            //--------------Planar-Boundary-Curves-----------------------//
+            count = checkedCurves.Count;
+            checkedCurves = checkedCurves.Where(x => x.IIsPlanar(tolerance)).ToList();
+            if (checkedCurves.Count != count)
+                Reflection.Compute.RecordWarning("Some of the input boundary curves were not planar within the tolerance and have been ignored on planar surface creation. Please make sure if the output is correct and tweak the input tolerance if needed.");
+
+            //--------------Closed-Boundary-Curves-----------------------//
+            count = checkedCurves.Count;
+            checkedCurves = checkedCurves.Where(x => x.IIsClosed(tolerance)).ToList();
+            if (checkedCurves.Count != count)
+                Reflection.Compute.RecordWarning("Some of the input boundary curves were not closed within the tolerance and have been ignored on planar surface creation. Please make sure if the output is correct and tweak the input tolerance if needed.");
+
+            if (checkedCurves.Count == 0)
+            {
+                Reflection.Compute.RecordError("Planar surface could not be created because all input boundary curves turned to be invalid (null, unsupported type, non-planar, not closed within the tolerance).");
```

Comment:
Wording of this error is a bit weird, suggest we change it to:
"Planar surface could not be created because all input boundary curves are invalid (null, unsupported type, non-planar, or not closed within the tolerance)."